database.hh

```diff
@@ -533,7 +533,7 @@ private:
     utils::phased_barrier _pending_streams_phaser;
```

Comment:
This commit seems to do more than what is in the description, it now also syncs the touched directories.

Reply:
Why? `sstable::move_to_new_dir_in_thread` used to do it unconditionally before this change.